cmd/containerboot/main.go

```diff
@@ -441,12 +459,32 @@ runLoop:
 					}
 					currentEgressIPs = newCurentEgressIPs
 				}
-				if cfg.ProxyTo != "" && len(addrs) > 0 && ipsHaveChanged {
+				if cfg.ProxyTargetIP != "" && len(addrs) > 0 && ipsHaveChanged {
 					log.Printf("Installing proxy rules")
-					if err := installIngressForwardingRule(ctx, cfg.ProxyTo, addrs, nfr); err != nil {
+					if err := installIngressForwardingRule(ctx, cfg.ProxyTargetIP, addrs, nfr); err != nil {
 						log.Fatalf("installing ingress proxy rules: %v", err)
 					}
 				}
+				if cfg.ProxyTargetDNSName != "" {
+					newBackendAddrs, err := resolveDNS(ctx, cfg.ProxyTargetDNSName)
+					if err != nil {
+						log.Printf("unable to resolve DNS name %s: %v, retrying in %s", cfg.ProxyTargetDNSName, err, proxyTargetIPsResolvePeriod)
```

Comment:
If we're unlucky and happen to hit a transient network issue on our first attempt to resolveDNS, we'll end up having to wait for 10 minutes before this starts working. It might be better to retry with an exponential backoff when we encounter an error.

Instead of doing this on a static ticker, what if we did something like this:

1. On initialization, perform the lookup and firewall configuration right away
2. Then, on every connection request, perform the DNS lookup again. If the IPs changed, reconfigure the firewall.

Yet another option would be to look at the TTL of the DNS responses and schedule the recheck/reconfigure based on that.

Reply:
> Then, on every connection request, perform the DNS lookup again

We don't have anything in code that is aware of connection requests and I am not sure whether the connection request/IP address change patterns would be such that this would actually be good user experience.

We _could_ reduce the period till the next lookup on a failure with an exponential backoff for repeated lookups post failures up till whatever is the static period. And yeah could replace the static 10 minutes with TTL 👍🏼

Reply:
> We don't have anything in code that is aware of connection requests

Oh yeah, good point!

> We could reduce the period till the next lookup on a failure with an exponential backoff for repeated lookups post failures up till whatever is the static period. And yeah could replace the static 10 minutes with TTL

Sounds good.

Reply:
@oxtoacart I ended up not using TTL - that would have needed pulliing in miekg/dns and I don't want to (yet) parse in the dns responses from there to extract TTL. I was playing around with our internal recursive.Resolver for this that returns a more suitable response, but that seems to return very unexpectedly short TTLs- I want to look more at what is the expected behaviour from that, but I think for this PR keeping the default and the max check period to 10 minutes could be good enough- we can see how well it works for users who want this.

I've added logic to start with shorter checks (from 2s to 10 minutes, exponentially upwards) for consecutive failing resolution checks.

Reply:
https://github.com/tailscale/tailscale/pull/11802/commits/e2a1b15777208754ef5e35561ef6d24c246f977f#diff-700830ac79a41da42cd3a82065f0fee3003a4390eead4db1de4bf30557a06bd6R390-R411